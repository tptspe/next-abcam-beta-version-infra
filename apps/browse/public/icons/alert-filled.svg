<svg xmlns="http://www.w3.org/2000/svg" width="24" height="25" fill="none" viewBox="0 0 24 25">
    <path fill="#273F3F" fill-rule="evenodd" d="M9.432 3.847l.003-.005C9.979 2.946 10.95 2.398 12 2.398c1.048 0 2.02.548 2.568 1.449l8.478 14.154c.533.923.536 2.06.008 2.986-.528.926-1.507 1.502-2.584 1.514H3.519c-1.066-.012-2.046-.588-2.574-1.514-.527-.926-.524-2.063.017-3l8.47-14.14zM12 18.501c.552 0 1-.448 1-1s-.448-1-1-1c-.553 0-1 .448-1 1s.447 1 1 1zm0-10c-.553 0-1 .448-1 1v4c0 .552.447 1 1 1 .552 0 1-.448 1-1v-4c0-.552-.448-1-1-1z" clip-rule="evenodd"/>
</svg>
