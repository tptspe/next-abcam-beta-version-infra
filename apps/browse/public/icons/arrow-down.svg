<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none" viewBox="0 0 24 24">
    <path fill="#273F3F" d="M11 17.826V4c0-.552.448-1 1-1s1 .448 1 1v13.826l5.35-4.585c.419-.36 1.05-.311 1.41.108.359.42.31 1.05-.11 1.41l-7 6c-.374.321-.926.321-1.3 0l-7-6c-.42-.36-.469-.99-.11-1.41.36-.42.991-.468 1.41-.108L11 17.826z"/>
</svg>
