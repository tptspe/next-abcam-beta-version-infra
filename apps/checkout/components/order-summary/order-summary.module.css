@media screen {
  .OrderSummary {
    @apply text-black;
    @apply m-auto;

    font-weight: 600;
  }

  .subTotal,
  .subTotalFolded {
    @apply flex;
    @apply justify-between;
    @apply pt-4;
    @apply pb-1.5;
  }

  .shippingAndHandling,
  .shippingAndHandlingFolded {
    @apply flex;
    @apply justify-between;
    @apply text-xs;
    @apply py-1.5;
  }

  .shippingAndHandlingHeader {
    height: 1.125rem;
  }

  .shippingAndHandlingValue {
    height: 1.125rem;
  }

  .shippingAndHandling .countrySelector,
  .shippingAndHandlingFolded .countrySelector {
    @apply bg-transparent;
    @apply text-buttonPrimary;
    @apply mx-0;
    @apply border-none;
    @apply pr-0;
    @apply py-0;

    span {
      @apply pr-0;
    }
  }

  .tax,
  .taxFolded {
    @apply flex;
    @apply justify-between;
    @apply text-xs;
    @apply py-1.5;
  }

  .taxHeader {
    @apply h-6;
  }

  .taxValue {
    @apply h-6;
  }

  div .middleLine,
  div .middleLineFolded {
    @apply my-1.5;
  }

  .total,
  .totalFolded {
    @apply flex;
    @apply justify-between;
    @apply py-1.5;
  }

  .totalHeader {
    height: 1.875rem;
    width: 216px;
  }

  .totalValue {
    height: 1.875rem;
  }

  .buttons,
  .buttonsFolded {
    @apply flex;
    @apply flex-col;
  }

  .loginButton,
  .loginButtonFolded {
    @apply block;
    @apply max-w-full;
    @apply self-stretch;
    @apply mb-3;
    @apply text-center;

    margin-top: 1.125rem;
  }

  .continueButton,
  .continueButtonFolded {
    @apply bg-transparent;
    @apply flex;
    @apply border-transparent;
    @apply mt-3;
    @apply mb-4;
    @apply mx-auto;
  }

  @media (max-width: 768px) {
    .content {
      @apply bg-grey95;
      @apply p-5;
      @apply rounded;
    }

    .contentFolded {
      @apply px-10;
      @apply pb-5;
      @apply bg-white;
    }

    .subTotalFolded {
      @apply mt-0;
      @apply justify-around;
    }

    .shippingAndHandlingFolded {
      @apply hidden;
    }

    .taxFolded {
      @apply hidden;
    }

    div .middleLineFolded {
      @apply hidden;
    }

    .totalFolded {
      @apply hidden;
    }

    .loginButtonFolded {
      @apply mt-0.5;
      @apply mb-0;
      @apply justify-around;
    }

    .continueButtonFolded {
      @apply hidden;
    }

    @media (max-width: 640px) {
      .content {
        @apply rounded-none;
      }

      @media (max-width: 475px) {
        .continueButton {
          @apply mb-0;
        }
      }
    }
  }
}
