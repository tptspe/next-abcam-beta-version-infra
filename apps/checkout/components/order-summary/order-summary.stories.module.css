@media screen {
  .shoppingBasket {
    @apply py-2;
    @apply h-auto;
    @apply flex;
    @apply flex-col;
    @apply px-20;
  }

  .header {
    @apply text-grey20;
    @apply py-5;
    @apply mb-2;
    @apply border-b;

    font-weight: 600;
  }

  .backButton {
    @apply flex;

    font-size: 12px;
  }

  .title {
    @apply mt-1;

    font-size: 24px;
  }

  .content {
    @apply flex;
    @apply flex-col;
  }

  .ico {
    @apply pt-1;
  }

  .orderSummaryContainer,
  .orderSummaryContainerScrolled {
    @apply order-none;
  }

  .itemsContainer {
    @apply order-none;
  }

  @media (max-width: 768px) {
    .shoppingBasket {
      @apply px-10;
    }

    .header {
      @apply py-0;
      @apply mb-1;
      @apply z-10;
      @apply bg-white;
      @apply pt-5;
      @apply pb-2;
      @apply sticky;
      @apply top-0;
    }

    .orderSummaryContainer {
      @apply order-none;
      @apply sticky;
      @apply w-full;
      @apply p-5;
      @apply px-24;
    }

    .orderSummaryContainerScrolled {
      @apply fixed;
      @apply w-full;
      @apply mt-0;
      @apply border-t;
      @apply shadow-elevation-1;

      z-index: 10;
      left: 0;
      top: 84px;
    }
  }

  @media (max-width: 640px) {
    .header {
      padding-left: 40px;

      @apply z-10;
      @apply bg-white;
      @apply mb-4;
      @apply py-2;
      @apply sticky;
      @apply top-0;
      @apply border-b;
    }

    .orderSummaryContainer {
      @apply order-none;
      @apply self-center;
      @apply sticky;
      @apply w-full;
      @apply p-0;

      top: 74px;
    }

    .orderSummaryContainerScrolled {
      top: 72px;
    }
  }

  @media (max-width: 475px) {
    .shoppingBasket {
      @apply px-0;
    }

    .content {
      @apply pt-0;
    }

    .header {
      @apply z-10;
      @apply bg-white;
      @apply sticky;
      @apply top-0;
      @apply mb-0;
      @apply pl-6;
      @apply pb-16;

      height: 80px;
    }

    .itemsContainer {
      padding-left: 1.875rem;
      padding-right: 1.875rem;
    }

    .orderSummaryContainer {
      @apply order-none;
      @apply self-center;
      @apply sticky;
      @apply w-full;

      top: 80px;
    }
  }
}
