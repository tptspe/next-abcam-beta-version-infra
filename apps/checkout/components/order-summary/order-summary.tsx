import { Button } from '@abcam-web/lego-shared-components/lib/button/button'
import React, { FC } from 'react'
import classNames from 'classnames'

import styles from './order-summary.module.css'
import type { Price } from '@checkout-shared/types'
export interface OrderSummaryProps {
  itemsCount: number
  subtotal: Price
  vat: Price
  vatPercentage: number
  total: Price
  isLoading?: boolean
  isExpanded: boolean
  loginOnClick: () => void
  continueOnClick: () => void
  chooseCountryOnClick: () => void
}

const OrderSummary: FC<OrderSummaryProps> = ({
  itemsCount,
  subtotal,
  vat,
  vatPercentage,
  total,
  isExpanded,
  isLoading,
  loginOnClick,
  continueOnClick,
  chooseCountryOnClick,
}) => {
  return (
    <div className={styles.OrderSummary}>
      {!isLoading && (
        <div
          className={classNames({
            [styles.content]: isExpanded,
            [styles.contentFolded]: !isExpanded,
          })}
        >
          <div
            className={classNames({
              [styles.subTotal]: isExpanded,
              [styles.subTotalFolded]: !isExpanded,
            })}
          >
            <div
              className={classNames({
                [styles.subTotalHeader]: isExpanded,
                [styles.subTotalHeaderFolded]: !isExpanded,
              })}
            >
              <p>Subtotal ({itemsCount} items)</p>
            </div>
            <div
              className={classNames({
                [styles.subTotalValue]: isExpanded,
                [styles.subTotalValueFolded]: !isExpanded,
              })}
            >
              <p>
                {subtotal.currency}
                {subtotal.value}
              </p>
            </div>
          </div>
          <div
            className={classNames({
              [styles.shippingAndHandling]: isExpanded,
              [styles.shippingAndHandlingFolded]: !isExpanded,
            })}
          >
            <div className={styles.shippingAndHandlingHeader}>
              <p>Shipping &amp; handling</p>
            </div>
            <div className={styles.shippingAndHandlingValue}>
              <Button
                as="a"
                className={classNames(
                  'whitespace-nowrap',
                  styles.countrySelector
                )}
                onClick={chooseCountryOnClick}
                variant="primary"
                size="small"
              >
                Choose your country to estimate
              </Button>
            </div>
          </div>
          <div
            className={classNames({
              [styles.tax]: isExpanded,
              [styles.taxFolded]: !isExpanded,
            })}
          >
            <div className={styles.taxHeader}>
              <p>VAT ({vatPercentage}%)</p>
            </div>
            <div className={styles.taxValue}>
              <p>
                {vat.currency}
                {vat.value}
              </p>
            </div>
          </div>
          <div>
            <hr
              className={classNames({
                [styles.middleLine]: isExpanded,
                [styles.middleLineFolded]: !isExpanded,
              })}
            />
          </div>
          <div
            className={classNames({
              [styles.total]: isExpanded,
              [styles.totalFolded]: !isExpanded,
            })}
          >
            <div className={styles.totalHeader}>
              <p>Order total</p>
            </div>
            <div className={styles.totalValue}>
              <p>
                {total.currency}
                {total.value}
              </p>
            </div>
          </div>
          <div
            className={classNames({
              [styles.buttons]: isExpanded,
              [styles.buttonsFolded]: !isExpanded,
            })}
          >
            <Button
              className={classNames({
                'whitespace-nowrap': true,
                [styles.loginButton]: isExpanded,
                [styles.loginButtonFolded]: !isExpanded,
              })}
              onClick={loginOnClick}
              variant="primary"
              size="large"
            >
              Login to checkout
            </Button>
            <Button
              className={classNames({
                'whitespace-nowrap': true,
                [styles.continueButton]: isExpanded,
                [styles.continueButtonFolded]: !isExpanded,
              })}
              onClick={continueOnClick}
              variant="tertiary"
              size="large"
            >
              Continue browsing
            </Button>
          </div>
        </div>
      )}
    </div>
  )
}

export { OrderSummary }
