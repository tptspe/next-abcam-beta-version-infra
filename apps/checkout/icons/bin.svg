<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="none" viewBox="0 0 20 20">
    <path fill="#D43737" fill-rule="evenodd" d="M7.5 4.167h5v-.834c0-.46-.373-.833-.833-.833H8.334c-.46 0-.834.373-.834.833v.834zm6.667 0v-.834c0-1.38-1.12-2.5-2.5-2.5H8.334c-1.381 0-2.5 1.12-2.5 2.5v.834H2.5c-.46 0-.833.373-.833.833 0 .46.373.833.833.833h.834v10.834c0 1.38 1.119 2.5 2.5 2.5h8.333c1.38 0 2.5-1.12 2.5-2.5V5.833h.833c.46 0 .834-.373.834-.833 0-.46-.373-.833-.834-.833h-3.333zm-.833 1.666H5v10.834c0 .46.373.833.834.833h8.333c.46 0 .833-.373.833-.833V5.833h-1.666zM7.5 9.167c0-.46.373-.834.834-.834.46 0 .833.373.833.834v5c0 .46-.373.833-.833.833-.46 0-.834-.373-.834-.833v-5zm4.167-.834c-.46 0-.833.373-.833.834v5c0 .46.373.833.833.833.46 0 .833-.373.833-.833v-5c0-.46-.373-.834-.833-.834z" clip-rule="evenodd"/>
</svg>
