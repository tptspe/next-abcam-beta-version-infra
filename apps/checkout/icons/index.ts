export { ReactComponent as Bin } from './bin.svg'
