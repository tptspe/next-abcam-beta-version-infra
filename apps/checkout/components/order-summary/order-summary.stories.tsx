import React, { FC, useEffect, useState } from 'react'
import { OrderSummary, OrderSummaryProps } from './order-summary'
import styles from './order-summary.stories.module.css'
import classNames from 'classnames'

export default {
  component: OrderSummary,
  title: 'OrderSummary',
}

const OrderSummaryExample: FC = () => {
  const [isExpanded, setIsExpanded] = useState(true)
  useEffect(() => {
    if (typeof window !== 'undefined') {
      window.onscroll = () => {
        const currentScrollPos = window.pageYOffset
        if (currentScrollPos > 0) {
          setIsExpanded(false)
        } else {
          setIsExpanded(true)
        }
      }
    }
  }, [])

  const props: OrderSummaryProps = {
    itemsCount: 4,
    subtotal: { value: 1800, currency: '£' },
    vat: { value: 360, currency: '£' },
    vatPercentage: 20,
    total: { value: 2176, currency: '£' },
    isExpanded: isExpanded,
    loginOnClick: (): void => {
      console.log('Login clicked')
    },
    continueOnClick: (): void => {
      console.log('Continue clicked')
    },
    chooseCountryOnClick: (): void => {
      console.log('Choose country clicked')
    },
  }

  return (
    <>
      <div className={styles.shoppingBasket}>
        <div className={styles.content}>
          <div
            className={classNames({
              [styles.orderSummaryContainer]: isExpanded,
              [styles.orderSummaryContainerScrolled]: !isExpanded,
            })}
          >
            <OrderSummary {...props} />
          </div>
          <p style={{ height: '300vh' }} />
        </div>
      </div>
    </>
  )
}

export const primary = () => {
  return <OrderSummaryExample />
}
