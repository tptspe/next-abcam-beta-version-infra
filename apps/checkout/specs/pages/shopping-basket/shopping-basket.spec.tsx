import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ShoppingBasket from '../../../pages/shopping-basket/shopping-basket'
import { ArrowLeft } from '@abcam-web/lego-shared-components/icons'
import Adapter from 'enzyme-adapter-react-16'
import { OrderSummary } from '@checkout/components/order-summary/order-summary'

Enzyme.configure({ adapter: new Adapter() })

describe('ShoppingBasket', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = shallow(<ShoppingBasket />)
  })

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should render shoppingBasket correctly', () => {
    const shoppingBasket = wrapper.find('.shoppingBasket')
    expect(shoppingBasket).toHaveLength(1)
  })

  it('should render back button correctly', () => {
    const backButton = wrapper.find('.backButton')
    expect(backButton).toHaveLength(1)
    expect(backButton.html()).toContain('Back')
    const arrowLeft = wrapper.find(ArrowLeft)
    expect(arrowLeft).toHaveLength(1)
  })

  it('should render title correctly', () => {
    const title = wrapper.find('.title')
    expect(title).toHaveLength(1)
  })

  it('should render BasketItemList correctly', () => {
    const content = wrapper.find('.content')
    expect(content).toHaveLength(1)
    const basketItemList = wrapper.find('BasketItemList')
    expect(basketItemList).toHaveLength(1)
  })

  it('should render OrderSummary correctly', () => {
    const orderSummaryContainer = wrapper.find('.orderSummaryContainer')
    expect(orderSummaryContainer).toHaveLength(1)
    const orderSummary = wrapper.find('OrderSummary')
    expect(OrderSummary).toHaveLength(1)
  })
})
