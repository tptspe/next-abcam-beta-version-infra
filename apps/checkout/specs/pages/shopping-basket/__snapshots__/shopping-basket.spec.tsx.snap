// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingBasket should match the snapshot 1`] = `
<div
  className="shoppingBasket"
>
  <div
    className="header"
  >
    <div
      className="backButton"
    >
      <ForwardRef(SvgArrowLeft)
        className="ico"
      />
       Back
    </div>
    <div
      className="title"
    >
      Shopping Basket
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="itemsContainer"
    >
      <BasketItemList
        backURL="something"
        deleteOnClick={[Function]}
        isLoading={false}
        items={Array []}
        quantityOnChange={[Function]}
        sizeOnChange={[Function]}
      />
    </div>
    <div
      className="orderSummaryContainer"
    >
      <OrderSummary
        chooseCountryOnClick={[Function]}
        continueOnClick={[Function]}
        isExpanded={true}
        isLoading={false}
        itemsCount={0}
        loginOnClick={[Function]}
        subtotal={
          Object {
            "currency": "€",
            "value": 0,
          }
        }
        total={
          Object {
            "currency": "€",
            "value": 0,
          }
        }
        vat={
          Object {
            "currency": "€",
            "value": 0,
          }
        }
        vatPercentage={0}
      />
    </div>
  </div>
</div>
`;
